src/watertap_contrib/seto/costing/units/chemical_softening_0D.py

```diff
@@ -0,0 +1,826 @@
+        expr=blk.admin_capital_cost
+        == 69195
+        * (
+            (
+                pyunits.convert(
+                    blk.unit_model.properties_in[0].flow_vol,
+                    to_units=pyunits.m**3 / pyunits.day,
+                )
+                / 3785
+            )
+            ** 0.5523
+        )
+    )
+
+    # endregion
+
+    # region Operational cost component constraints
+
+    # Mixing tank
+    if value(blk.unit_model.vel_gradient_mix) <= 300:
+        blk.mix_tank_op_cost_constraint = Constraint(
+            expr=blk.mix_tank_op_cost
+            == (
+                -3e-8
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                ** 3
+            )
+            + (
+                0.0008
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                ** 2
+            )
+            + (
+                2.8375
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+            )
+            + 22588
+        )
+
+    elif (
+        value(blk.unit_model.vel_gradient_mix) > 300
+        and value(blk.unit_model.vel_gradient_mix) <= 600
+    ):
+        blk.mix_tank_op_cost_constraint = Constraint(
+            expr=blk.mix_tank_op_cost
+            == (
+                -3e-8
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                ** 3
+            )
+            + (
+                0.0008
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                ** 2
+            )
+            + (
+                7.8308
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                + 22588
+            )
+        )
+
+    elif (
+        value(blk.unit_model.vel_gradient_mix) > 600
+        and value(blk.unit_model.vel_gradient_mix) <= 1000
+    ):
+        blk.mix_tank_op_cost_constraint = Constraint(
+            expr=blk.mix_tank_op_cost
+            == (
+                36.096
+                * (
+                    pyunits.convert(
+                        blk.unit_model.volume_mixer, to_units=pyunits.ft**3
+                    )
+                )
+                ** 2
+                + 18928
+            )
+        )
+
+    else:
+        print("Error - Out of bounds (300 - 1000 s-1)")
```

Comment:
We would either use a logger error or warning, raise an exception, or both, but we want to avoid printing things like this (especially if meant to denote an error).